Extension/src/LanguageServer/extension.ts

```diff
@@ -1387,3 +1391,134 @@ export async function preReleaseCheck(): Promise<void> {
+    const hoverPosition = copilotHoverProvider.getCurrentHoverPosition();
+    if (!hoverDocument || !hoverPosition) {
+        return;
+    }
+
+    // Prep hover with wait message.
+    copilotHoverProvider.showWaiting();
+
+    if (copilotHoverProvider.isCancelled(hoverDocument, hoverPosition)) {
+        return;
+    }
+
+    // Move the cursor to the hover position, but don't focus the editor.
+    await vscode.window.showTextDocument(hoverDocument, { preserveFocus: true, selection: new vscode.Selection(hoverPosition, hoverPosition) });
+
+    if (!await showCopilotContent(copilotHoverProvider, hoverDocument, hoverPosition)) {
+        return;
+    }
+
+    // Gather the content for the query from the client.
+    const requestInfo = await copilotHoverProvider.getRequestInfo(hoverDocument, hoverPosition);
+    if (requestInfo.length === 0) {
+        await reportCopilotFailure(copilotHoverProvider, hoverDocument, hoverPosition, "Failed to receive request info from the client.");
+        return;
+    }
+
+    const locale = getLocaleId();
+
+    const messages = [
+        vscode.LanguageModelChatMessage
+            .User(requestInfo + locale)];
+
+    const [model] = await vscodelm.selectChatModels(modelSelector);
+
+    let chatResponse: vscode.LanguageModelChatResponse | undefined;
+    try {
+        chatResponse = await model.sendRequest(
+            messages,
+            {},
+            copilotHoverProvider.getCurrentHoverCancellationToken()
+        );
+    } catch (err) {
+        if (err instanceof vscode.LanguageModelError) {
+            console.log(err.message, err.code, err.cause);
+            await reportCopilotFailure(copilotHoverProvider, hoverDocument, hoverPosition, err.message);
+        } else {
+            throw err;
+        }
+        return;
+    }
+
+    // Ensure we have a valid response from Copilot.
+    if (!chatResponse) {
+        await reportCopilotFailure(copilotHoverProvider, hoverDocument, hoverPosition);
+        return;
+    }
+
+    let content: string = '';
+
+    try {
+        for await (const fragment of chatResponse.text) {
+            content += fragment;
+        }
+    } catch (err) {
+        if (err instanceof Error) {
+            console.log(err.message, err.cause);
+            await reportCopilotFailure(copilotHoverProvider, hoverDocument, hoverPosition, err.message);
+        }
+        return;
+    }
+
+    if (content.length === 0) {
+        await reportCopilotFailure(copilotHoverProvider, hoverDocument, hoverPosition);
+        return;
+    }
+
+    await showCopilotContent(copilotHoverProvider, hoverDocument, hoverPosition, content);
+}
+
+async function reportCopilotFailure(copilotHoverProvider: CopilotHoverProvider, hoverDocument: vscode.TextDocument, hoverPosition: vscode.Position, customError?: string): Promise<void> {
+    const errorMessage = customError ? customError : "";
+    telemetry.logLanguageServerEvent("CopilotHoverError", { "ErrorMessage": errorMessage });
```

Comment:
Could you log a separate error telemetry when there's no declaration/definition? I don't think that's really an error since it seems "by design" when used on a macro, etc. It seems like the user facing message should also not be "An error occurred while generating Copilot summary." and instead say the something that sounds more "by design" and less like an unexpected error occurred?

Reply:
![image](https://github.com/user-attachments/assets/bc1eb2a1-c106-4e3b-979a-b5920c1cf761)

Reply:
The telemetry and/or message could have an issue filed to follow up too -- up to your team and if they care about no definition/declaration being counted as "errors".

Reply:
Umm...but it would be easier to triage user reported "bugs" if known/"by design" issues had a different message so that we don't have to investigate the "error".

Reply:
Changed this case where we have no decl/defn context to not log an error and return a more descriptive message letting the user know that it's not available. I also updated the error logging to require an error message to include in the logging, even though we'll still show the generic error occurred message to the user, but it shouldn't show up in the telemetry.